packages/loader/container-loader/src/container.ts

```diff
@@ -1453,6 +1465,21 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private processRemoteMessage(message: ISequencedDocumentMessage): IProcessMessageResult {
         const local = this._clientId === message.clientId;
 
+         // Report if we're getting messages from a clientId that we previously marked inactive
+         if (!local) {
+            const client: IActiveSequencedClient | undefined =
+                this._protocolHandler?.quorum.getMember(message.clientId);
+            if (client?.isActive === false) {
```

Comment:
this won't catach the issue. both clients were still active from the servers persepctive which is what quorum represents

Reply:
if a client id is not active from the server's perspective, i'm pretty sure the server will nack any ops for that client, so this should also never happen

Reply:
Isn't this what we're looking for though?  (Cases where these ops should have been nacked, but aren't)
The `isActive` flag is set locally on our side, so the only input the server has on it here is if the client gets removed from the quorum.  This is the same place we added telemetry before, except this time the quorum is being used to remove clientIds that should no longer matter instead of keeping them indefinitely until they are bumped out of the MRU.

Reply:
No. That's not what happens. The old client id has not had a leave message sent by the server, as it was the client who decided it needed to reconnect. We see the first set of ops from the old client id, then the resubmitted ops from the new client id, and only after that do we see a leave message for the old client id.

Reply:
i updated #3627 to make this explicit

Reply:
Here are the steps as I understand and how this PR fits into it, which I _think_ matches what's in #3627.  If you could help me with which part I'm missing, I'd appreciate it.

1. Old client join message
2. First set of ops sent from old client id
3. New client join message
3a. During new client connect, we hit the change in L1398 above.  When we change our client id, we see the old client is still in the quorum (because there has not been an old client leave message yet), so we mark it as `isActive = false`, but don't remove it from the quorum.  
4. Second set of ops sent from old client id
4a. New client sees second set of ops and hits the change here.  It doesn't recognize them as coming from the same client (`!local`), but because old client has not had its leave message yet (and is still in the quorum), new client will find it in the quorum marked `!isActive` (which was done earlier on new client connect).  It then reports telemetry.
5. Duplicate second set of ops are sent from new client (because the ops sent from old client were not recognized)
6. Old client leave message

Reply:
Yep, that looks correct. :) There might be many reasons for the issue, but this check is equivalent to what an keeping at array of 100 clientIDs did - we should evaluate equivalence here only.
If there are more checks or more cases to track, we should do it in separate PR.